Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp

```diff
@@ -1057,50 +1057,47 @@ IPCCommandResult CWII_IPC_HLE_Device_net_ip_top::IOCtl(u32 _CommandAddress)
-
+				// Host name; located immediately after struct
+				static const u32 GETHOSTBYNAME_STRUCT_SIZE = 0x10;
 				u32 name_length = (u32)strlen(remoteHost->h_name) + 1;
-				Memory::CopyToEmu(wii_addr, remoteHost->h_name, name_length);
-				Memory::Write_U32(wii_addr, BufferOut);
-				wii_addr += (name_length + 4) & ~3;
-
-				// aliases - empty
-				Memory::Write_U32(wii_addr, BufferOut + 4);
-				Memory::Write_U32(wii_addr + sizeof(u32), wii_addr);
-				wii_addr += sizeof(u32);
-				Memory::Write_U32(0, wii_addr);
-				wii_addr += sizeof(u32);
-
-				// hardcode to ipv4
-				_dbg_assert_msg_(WII_IPC_NET,
-					remoteHost->h_addrtype == AF_INET && remoteHost->h_length == sizeof(u32),
-					"returned host info is not IPv4");
-				Memory::Write_U16(AF_INET, BufferOut + 8);
-				Memory::Write_U16(sizeof(u32), BufferOut + 10);
-
-				// addrlist - probably only really need to return 1 anyways...
-				Memory::Write_U32(wii_addr, BufferOut + 12);
-				u32 num_addr = 0;
-				while (remoteHost->h_addr_list[num_addr])
-					num_addr++;
-				for (u32 i = 0; i < num_addr; ++i)
+				Memory::CopyToEmu(BufferOut + GETHOSTBYNAME_STRUCT_SIZE, remoteHost->h_name, name_length);
+				Memory::Write_U32(BufferOut + GETHOSTBYNAME_STRUCT_SIZE, BufferOut);
+
+				// IP address list; located at offset 0x110.
+				static const u32 GETHOSTBYNAME_IP_LIST_OFFSET = 0x110;
+				u32 num_ip_addr = 0;
+				while (remoteHost->h_addr_list[num_ip_addr])
+					num_ip_addr++;
+				for (int i = 0; remoteHost->h_addr_list[i]; ++i)
 				{
-					Memory::Write_U32(wii_addr + sizeof(u32) * (num_addr + 1), wii_addr);
-					wii_addr += sizeof(u32);
+					u32 addr = BufferOut + GETHOSTBYNAME_IP_LIST_OFFSET + i * 4;
+					Memory::Write_U32_Swap(*(u32*)(remoteHost->h_addr_list[i]), addr);
 				}
-				// null-terminated list
-				Memory::Write_U32(0, wii_addr);
-				wii_addr += sizeof(u32);
-				// The actual IPs
-				for (int i = 0; remoteHost->h_addr_list[i]; ++i)
+
+				// List of pointers to IP addresses; located at offset 0x340.
+				// This must be exact: PPC code to convert the struct hardcodes
+				// this offset.
+				static const u32 GETHOSTBYNAME_IP_PTR_LIST_OFFSET = 0x340;
+				Memory::Write_U32(BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET, BufferOut + 12);
+				for (u32 i = 0; i < num_ip_addr; ++i)
 				{
-					Memory::Write_U32_Swap(*(u32*)(remoteHost->h_addr_list[i]), wii_addr);
-					wii_addr += sizeof(u32);
+					u32 addr = BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + i * 4;
+					Memory::Write_U32(BufferOut + GETHOSTBYNAME_IP_LIST_OFFSET + i * 4, addr);
 				}
+				Memory::Write_U32(0, BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4);
+
+				// Aliases - empty.
+				// TODO: Fill this in?  (Not that it's particularly useful...)
+				Memory::Write_U32(BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4, BufferOut + 4);
+				Memory::Write_U32(0, BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4 + 4);
+
```

Comment:
This second NULL terminator isn't necessary, h_aliases already points to the first one.